Source/DafnyCore/Backends/Dafny/AST.dfy

```diff
@@ -130,10 +130,30 @@ module {:extern "DAST"} DAST {
 
   datatype Primitive = Int | Real | String | Bool | Char | Native
 
-  // USIZE is for whatever target considers that native arrays can be indexed with
+
   datatype NewtypeRange =
-    | U8 | I8 | U16 | I16 | U32 | I32 | U64 | I64 | U128 | I128 | USIZE | BigInt
+    | U8(overflow: bool) // Whether arithmetic operations can overflow and wrap
+    | I8(overflow: bool)
+    | U16(overflow: bool)
+    | I16(overflow: bool)
+    | U32(overflow: bool)
+    | I32(overflow: bool)
+    | U64(overflow: bool)
+    | I64(overflow: bool)
+    | U128(overflow: bool)
+    | I128(overflow: bool)
+    | USIZE  // For whatever target considers that native arrays can be indexed with
+    | BigInt
+    | Bool
     | NoRange
+  {
+    predicate canOverflow() {
```

Comment:
Why lowercase?

Reply:
I should be more careful about casing, thanks for reporting.